File: RickAndMortyAPP/app/src/main/java/com/example/lpiem/rickandmortyapp/Character.java
package com.example.lpiem.rickandmortyapp;

import android.os.Parcel;
import android.os.Parcelable;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.util.List;

public class Character implements Parcelable
{

    @SerializedName("id")
    @Expose
    private Integer id;
    @SerializedName("name")
    @Expose
    private String name;
    @SerializedName("status")
    @Expose
    private String status;
    @SerializedName("species")
    @Expose
    private String species;
    @SerializedName("type")
    @Expose
    private String type;
    @SerializedName("gender")
    @Expose
    private String gender;
    //@SerializedName("origin")
    //@Expose
    //private Origin origin;
    //@SerializedName("location")
    //@Expose
    //private Location location;
    //@SerializedName("image")
    @Expose
    private String image;
    @SerializedName("episode")
    @Expose
    private List<String> episode = null;
    @SerializedName("url")
    @Expose
    private String url;
    @SerializedName("created")
    @Expose
    private String created;
    public final static Parcelable.Creator<Character> CREATOR = new Creator<Character>() {


        @SuppressWarnings({
                "unchecked"
        })
        public Character createFromParcel(Parcel in) {
            return new Character(in);
        }

        public Character[] newArray(int size) {
            return (new Character[size]);
        }

    }
            ;

    protected Character(Parcel in) {
        this.id = ((Integer) in.readValue((Integer.class.getClassLoader())));
        this.name = ((String) in.readValue((String.class.getClassLoader())));
        this.status = ((String) in.readValue((String.class.getClassLoader())));
        this.species = ((String) in.readValue((String.class.getClassLoader())));
        this.type = ((String) in.readValue((String.class.getClassLoader())));
        this.gender = ((String) in.readValue((String.class.getClassLoader())));
        //this.origin = ((Origin) in.readValue((Origin.class.getClassLoader())));
        //this.location = ((Location) in.readValue((Location.class.getClassLoader())));
        this.image = ((String) in.readValue((String.class.getClassLoader())));
        in.readList(this.episode, (java.lang.String.class.getClassLoader()));
        this.url = ((String) in.readValue((String.class.getClassLoader())));
        this.created = ((String) in.readValue((String.class.getClassLoader())));
    }

    public Character() {
    }

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getStatus() {
        return status;
    }

    public void setStatus(String status) {
        this.status = status;
    }

    public String getSpecies() {
        return species;
    }

    public void setSpecies(String species) {
        this.species = species;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public String getGender() {
        return gender;
    }

    public void setGender(String gender) {
        this.gender = gender;
    }

    //public Origin getOrigin() {return origin;}

//    public void setOrigin(Origin origin) {
//        this.origin = origin;
//    }
//
//    public Location getLocation() {
//        return location;
//    }
//
//    public void setLocation(Location location) {
//        this.location = location;
//    }

    public String getImage() {
        return image;
    }

    public void setImage(String image) {
        this.image = image;
    }

    public List<String> getEpisode() {
        return episode;
    }

    public void setEpisode(List<String> episode) {
        this.episode = episode;
    }

    public String getUrl() {
        return url;
    }

    public void setUrl(String url) {
        this.url = url;
    }

    public String getCreated() {
        return created;
    }

    public void setCreated(String created) {
        this.created = created;
    }

    public void writeToParcel(Parcel dest, int flags) {
        dest.writeValue(id);
        dest.writeValue(name);
        dest.writeValue(status);
        dest.writeValue(species);
        dest.writeValue(type);
        dest.writeValue(gender);
//        dest.writeValue(origin);
//        dest.writeValue(location);
        dest.writeValue(image);
        dest.writeList(episode);
        dest.writeValue(url);
        dest.writeValue(created);
    }

    public int describeContents() {
        return  0;
    }

}

File: RickAndMortyAPP/app/src/main/java/com/example/lpiem/rickandmortyapp/Character.kt
package com.example.lpiem.rickandmortyapp

import android.os.Parcel
import android.os.Parcelable

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName
import java.io.Serializable

class Character : Parcelable, Serializable {

    companion object {
        @JvmField val CREATOR: Parcelable.Creator<Character> = object : Parcelable.Creator<Character> {


            override fun createFromParcel(inside: Parcel): Character {
                return Character(inside)
            }

            override fun newArray(size: Int): Array<Character?> {
                return arrayOfNulls(size)
            }

        }
    }

    @SerializedName("id")
    @Expose
    var id: Int? = null
    @SerializedName("name")
    @Expose
    var name: String? = null
    @SerializedName("status")
    @Expose
    var status: String? = null
    @SerializedName("species")
    @Expose
    var species: String? = null
    @SerializedName("type")
    @Expose
    var type: String? = null
    @SerializedName("gender")
    @Expose
    var gender: String? = null
    //@SerializedName("origin")
    //@Expose
    //private Origin origin;
    //@SerializedName("location")
    //@Expose
    //private Location location;
    //@SerializedName("image")
    //public Origin getOrigin() {return origin;}

    //    public void setOrigin(Origin origin) {
    //        this.origin = origin;
    //    }
    //
    //    public Location getLocation() {
    //        return location;
    //    }
    //
    //    public void setLocation(Location location) {
    //        this.location = location;
    //    }

    @Expose
    var image: String? = null
    @SerializedName("episode")
    @Expose
    var episode: List<String>? = null
    @SerializedName("url")
    @Expose
    var url: String? = null
    @SerializedName("created")
    @Expose
    var created: String? = null

    constructor(inside: Parcel) {
        this.id = inside.readValue(Int::class.java.classLoader) as Int
        this.name = inside.readValue(String::class.java.classLoader) as String
        this.status = inside.readValue(String::class.java.classLoader) as String
        this.species = inside.readValue(String::class.java.classLoader) as String
        this.type = inside.readValue(String::class.java.classLoader) as String
        this.gender = inside.readValue(String::class.java.classLoader) as String
        //this.origin = ((Origin) inside.readValue((Origin.class.getClassLoader())));
        //this.location = ((Location) inside.readValue((Location.class.getClassLoader())));
        this.image = inside.readValue(String::class.java.classLoader) as String
        inside.readList(this.episode, java.lang.String::class.java.classLoader)
        this.url = inside.readValue(String::class.java.classLoader) as String
        this.created = inside.readValue(String::class.java.classLoader) as String
    }

        override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeValue(id)
        dest.writeValue(name)
        dest.writeValue(status)
        dest.writeValue(species)
        dest.writeValue(type)
        dest.writeValue(gender)
        //        dest.writeValue(origin);
        //        dest.writeValue(location);
        dest.writeValue(image)
        dest.writeList(episode)
        dest.writeValue(url)
        dest.writeValue(created)
    }

    override fun describeContents(): Int {
        return 0
    }



}